src/cplscheme/MultiCouplingScheme.cpp

```diff
@@ -142,12 +198,19 @@ void MultiCouplingScheme::exchangeSecondData()
     }
   } else {
     receiveConvergence(_m2ns[_controller]);
-
     for (auto &receiveExchange : _receiveDataVector) {
       receiveData(_m2ns[receiveExchange.first], receiveExchange.second);
     }
     checkDataHasBeenReceived();
   }
+  if (hasConverged()) {
+    for (const DataMap::value_type &data : getAllData()) {
+      data.second->moveTimeStepsStorage();
+    }
```

Comment:
Here, we call for `getAllData()` the function `moveTimeStepsStorage()`. This performs extrapolation to compute an initial guess for all data (not only acceleration data). We have to call `moveTimeStepsStorage()` for all data anyway. Differentiating between accelerated data (use extrapolation when moving) and data that is not accelerated (don't use extrapolation when moving) would be overly complicated.